src/devices/Ili934x/Chsc6440.cs

```diff
@@ -0,0 +1,410 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Device.Gpio;
+using System.Device.I2c;
+using System.Drawing;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Iot.Device.Ili934x
+{
+    /// <summary>
+    /// Binding for Chipsemi CHSC6540 capacitive touch screen controller
+    /// Used for instance on the M5Tough in conjunction with an ILI9342 display controller.
+    /// Note: The M5Core2, while being very similar to the M5Tough otherwise, has a FT6336U instead.
+    /// The two chips appear to be similar, but the documentation is poor.
+    /// </summary>
+    public class Chsc6440 : IDisposable
```

Comment:
can the touch screen be added separately?

Reply:
Sure, but I really wouldn't prolong this even further.